### PR TITLE
feat: adding exit survey to confirmation pages (CMC-1117) (2/2)

### PR DESCRIPTION
**Pull request by devmarcy**
## JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CMC-1117

### Change description ###
Added exit survey to following events:

- Issue claim
- Retry claim
- Notify claim
- Notify claim details
- Extension Request 
- Respond
- Confirm to proceed
- Add or Amend claim documents
- Acknowledge Claim

Example of change (This is on Issue Claim event)

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/AddOrAmendClaimDocumentsCallbackHandler.java [hunk 1]
```diff
@@ -39,12 +39,17 @@
     }
 
     private SubmittedCallbackResponse buildConfirmation(CallbackParams callbackParams) {
+        String surveyLink = "https://www.smartsurvey.co.uk/s/CivilDamages_ExitSurvey_Claimant/";
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/AddOrAmendClaimDocumentsCallbackHandler.java]: how about making this configurable n yaml file?

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/NotifyClaimCallbackHandler.java [hunk 2]
```diff
@@ -35,8 +36,10 @@
     private static final List<CaseEvent> EVENTS = Collections.singletonList(NOTIFY_DEFENDANT_OF_CLAIM);
     public static final String CONFIRMATION_SUMMARY = "<br />The defendant legal representative's organisation has "
         + "been notified and granted access to this claim.\n\n"
-        + "You must notify the defendant with the claim details by %s";
+        + "You must notify the defendant with the claim details by %s"
+        + "\n\n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.";
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/NotifyClaimCallbackHandler.java]: please use %n instead

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/NotifyClaimDetailsCallbackHandler.java [hunk 3]
```diff
@@ -38,8 +39,10 @@
     private static final List<CaseEvent> EVENTS = Collections.singletonList(NOTIFY_DEFENDANT_OF_CLAIM_DETAILS);
     private static final String CONFIRMATION_SUMMARY = "<br />The defendant legal representative's organisation has"
         + " been notified of the claim details.\n\n"
-        + "They must respond by %s. Your account will be updated and you will be sent an email.";
+        + "They must respond by %s. Your account will be updated and you will be sent an email."
+        + "\n\n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.";
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/NotifyClaimDetailsCallbackHandler.java]: please use %n instead

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandlerTest.java [hunk 4]
```diff
@@ -91,10 +93,11 @@
         + " until payment is confirmed."
         + " Once payment is confirmed you will receive an email. The claim will then progress offline."
         + "\n\n To continue the claim you need to send the <a href=\"%s\" target=\"_blank\">sealed claim form</a>, "
-        + "a <a href=\"%s\" target=\"_blank\">response pack</a> and any supporting documents to "
+        + "a <a href=\"%3$s\" target=\"_blank\">response pack</a> and any supporting documents to "
         + "the defendant within 4 months. "
         + "\n\nOnce you have served the claim, send the Certificate of Service and supporting documents to the County"
-        + " Court Claims Centre.";
+        + " Court Claims Centre."
+        + "\n\n<br/><br/>This is a new service - your <a href=\"%2$s\" target=\"_blank\">feedback</a> will help us to improve it.";
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandlerTest.java]: please use %n instead

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/NotifyClaimCallbackHandlerTest.java [hunk 5]
```diff
@@ -114,7 +120,8 @@ void shouldSetClaimDetailsNotificationAsClaimNotificationDeadline_when14DaysIsSa
 
         private static final String CONFIRMATION_BODY = "<br />The defendant legal representative's organisation has "
             + "been notified and granted access to this claim.\n\n"
-            + "You must notify the defendant with the claim details by %s";
+            + "You must notify the defendant with the claim details by %s"
+            + "\n\n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.";
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/NotifyClaimCallbackHandlerTest.java]: please use %n instead

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java [hunk 6]
```diff
@@ -68,17 +69,20 @@
         + "receive an email. The email will also include the date when you need to notify the defendant of the claim."
         + "\n\n You must notify the defendant of the claim within 4 months of the claim being issued. The exact "
         + "date when you must notify the claim details will be provided when you first notify "
-        + "the defendant of the claim.";
+        + "the defendant of the claim."
+        + "\n\n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.";
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java]: please use %n instead

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java [hunk 7]
```diff
@@ -274,14 +278,17 @@ private String getHeader(CaseData caseData) {
...
 
         return format(
             caseData.getRespondent1Represented() == NO || caseData.getRespondent1OrgRegistered() == NO
                 ? LIP_CONFIRMATION_BODY
                 : CONFIRMATION_SUMMARY,
             format("/cases/case-details/%s#CaseDocuments", caseData.getCcdCaseReference()),
+            exitSurveyConfiguration.getClaimantSurvey(),
             claimIssueConfiguration.getResponsePackLink(),
             formattedServiceDeadline
+
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java]: unnecessary empty line

**Review by sabahirfan — DISMISSED** [review]

**Review by sabahirfan — DISMISSED** [review]

**Review by dharmendrak — DISMISSED** [review]

**Review by dharmendrak — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/config/ExitSurveyConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+@Data
+@Configuration
+public class ExitSurveyConfiguration {
+
+    private final String claimantSurvey;
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/config/ExitSurveyConfiguration.java]: to be consistent let's call it `applicantSurvey`

File: src/main/java/uk/gov/hmcts/reform/unspec/config/ExitSurveyConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    private final String defendantSurvey;
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/config/ExitSurveyConfiguration.java]: to be consistent let's call it `respondentSurvey`

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/AcknowledgeClaimCallbackHandler.java [hunk 10]
```diff
@@ -38,8 +39,10 @@
     private static final List<CaseEvent> EVENTS = Collections.singletonList(ACKNOWLEDGE_CLAIM);
 
     public static final String CONFIRMATION_SUMMARY = "<br />You need to respond to the claim before %s."
-        + "\n\n[Download the Acknowledgement of Claim form](%s)";
+        + "%n%n[Download the Acknowledgement of Claim form](%s)"
+        + "%n%n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.";
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/AcknowledgeClaimCallbackHandler.java]: @Service
class ExitSurveyContentGenerator {
    
 private final ExitSurveyConfiguration exitSurveyConfiguration;

public String applicantSurvey() {
return String.format("%n%n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.", exitSurveyConfiguration.getRespondentSurvey());
}
public String respondentSurvey() {
....
}
}

File: src/main/java/uk/gov/hmcts/reform/unspec/service/ExitSurveyContentService.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.unspec.config.ExitSurveyConfiguration;
+
+@Service
+@RequiredArgsConstructor
+public class ExitSurveyContentService {
+
+    private final ExitSurveyConfiguration exitSurveyConfiguration;
+    static final String feedbackLink = "%n%n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.";
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/service/ExitSurveyContentService.java]:
```suggestion
    private static final String feedbackLink = "%n%n<br/><br/>This is a new service - your <a href=\"%s\" target=\"_blank\">feedback</a> will help us to improve it.";
```

File: src/main/java/uk/gov/hmcts/reform/unspec/service/ExitSurveyContentService.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+    public String applicantSurvey() {
+
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/service/ExitSurveyContentService.java]: please remove this extra line

File: src/main/resources/application.yaml [hunk 13]
```diff
@@ -112,6 +112,11 @@ launchdarkly:
 testing:
   support:
     enabled: ${TESTING_SUPPORT_ENABLED:false}
+
+exitsurvey:
+  claimant: "https://www.smartsurvey.co.uk/s/CivilDamages_ExitSurvey_Claimant/"
```
Comment (sabahirfan) [inline src/main/resources/application.yaml]:
```suggestion
  applicant: "https://www.smartsurvey.co.uk/s/CivilDamages_ExitSurvey_Claimant/"
```

File: src/main/resources/application.yaml [hunk 14]
```diff
@@ -112,6 +112,11 @@ launchdarkly:
...
+  defendant: "https://www.smartsurvey.co.uk/s/CivilDamages_ExitSurvey_Defendant/"
```
Comment (sabahirfan) [inline src/main/resources/application.yaml]:
```suggestion
  respondent: "https://www.smartsurvey.co.uk/s/CivilDamages_ExitSurvey_Defendant/"
```

**Review by dharmendrak — DISMISSED** [review]
LGTM, please some comments of Sabah

**Review by sabahirfan — DISMISSED** [review]

**Review by sabahirfan — APPROVED** [review]
LGTM 👍

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**